trojsten/polls/templatetags/polls_parts.py

```diff
@@ -0,0 +1,72 @@
+from django import template
+from django.conf import settings
+from django.utils import timezone
+
+register = template.Library()
+
+
+@register.inclusion_tag("trojsten/polls/parts/progress.html", takes_context=True)
+def show_time(context, current_question):
+    start = current_question.created_date
+    end = current_question.deadline
+    full = end - start
+    remaining = end - timezone.now()
+    elapsed = full - remaining
+    if remaining.days <= settings.ROUND_PROGRESS_DANGER_DAYS:
+        progressbar_class = settings.ROUND_PROGRESS_DANGER_CLASS
+    elif remaining.days <= settings.ROUND_PROGRESS_WARNING_DAYS:
+        progressbar_class = settings.ROUND_PROGRESS_WARNING_CLASS
+    else:
+        progressbar_class = settings.ROUND_PROGRESS_DEFAULT_CLASS
+    percent = 100 * elapsed.days // full.days if full.days > 0 else 100
+    print("Percent is", percent, full.days, elapsed.days)
+    context.update(
+        {
+            "current": current_question,
+            "start": start,
+            "end": end,
+            "full": full,
+            "remaining": remaining,
+            "elapsed": elapsed,
+            "percent": percent,
+            "progressbar_class": progressbar_class,
+        }
+    )
+    return context
+
+
+def get_type(number):
+    if number == 1:
+        return 0
+    elif 2 <= number <= 4:
+        return 1
+    else:
+        return 2
+
+
+@register.filter
+def progress_time(delta):
+    if delta.days >= 1:
+        count = delta.days
+        return str(count) + " " + ["deň", "dni", "dní"][get_type(count)]

```

Comment:
Pouzi prekladanie + pluralizaciu namiesto custom logiky.

https://docs.djangoproject.com/en/3.0/topics/i18n/translation/#pluralization

Reply:
Vsimol som si, ze napriklad v ulohach na stranke je v progressbare 20 days, po anglicky, a cely zvysok stranky je po slovensky a tomu som sa chcel vyhnut. Chceme to naozaj tak?

Reply:
podla mna ta pluralizacia v gettext by to mala zvladnut spravne prelozit.

Reply:
Btw,  myslim ze na stranke len chybaju preklady, skusim sa na to pozriet niekedy. Volakedy to bolo poslovensky, len sa nam pri nejakom mergovani postracali nejake preklady.